% !TeX spellcheck = en_US
% !TeX encoding = UTF-8
\documentclass{beamer}

\mode<presentation> { \usetheme{Madrid} }

\usepackage{graphicx, graphics}
\usepackage[notocbib]{apacite}
\usepackage[style=iso]{datetime2}
\usepackage{enumerate}
\DeclareGraphicsExtensions{.pdf, .png, .jpg, .gif}

\AtBeginSection[]
{
    \begin{frame}
        \vfill
        \centering
        \begin{beamercolorbox}[sep=8pt,center,shadow=true,rounded=true]{title}
            \usebeamerfont{title}
            \insertsectionhead
            \par
        \end{beamercolorbox}
        \vfill
    \end{frame}
}

\AtBeginSubsection[]
{
    \begin{frame}
        \vfill
        \centering
        \begin{beamercolorbox}[sep=8pt, center, shadow=true, rounded=true]{title}
            \usebeamerfont{title}
            \insertsectionhead
            \par
        \end{beamercolorbox}
        \begin{beamercolorbox}[sep=4pt, center, shadow=true, rounded=true]{title}
            \usebeamerfont{subtitle}
            \insertsubsectionhead
            \par
        \end{beamercolorbox}
        \vfill
    \end{frame}
}

\title[Premature]{Metagenome Analysis of Premature Birth}

\author[Jaewoong Lee]
{
    Jaewoong Lee
    \and
    Semin Lee
}

\institute[UNIST BME]
{
    Department of Biomedical Engineering
    \newline
    Ulsan National Institute of Science and Technology
    \medskip
    \newline
    \textit{jwlee230@unist.ac.kr}
}

\date{\today}

\begin{document}
    \begin{frame}
        \titlepage
    \end{frame}

	\begin{frame}
        \frametitle{Overview}
        \tableofcontents[hideallsubsections]
    \end{frame}

    \section{Introduction}
    \begin{frame}
        \frametitle{Microbiome}

        \begin{itemize}
            \item Microbiota: the microorganisms which live inside \& on humans \cite{micro1}
            \item Microbiome: $10^{13}$ to $10^{14}$ microorganisms whose which collective genome \cite{micro2}
        \end{itemize}

        \begin{figure}[h!]
            \includegraphics[width=0.3 \linewidth]{figures/microbiome.jpg}
            \caption{Concept of a core human microbiome \protect \cite{micro1}}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{rRNA}

        \begin{itemize}
            \item Ribosomal RNA
            \item Well-known as a key to phylogeny \cite{rrna1}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Premature Birth (Preterm Birth)}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/premature.png}
            \caption{Definitions of Premature \protect\cite{premature1}}
        \end{figure}

        $\therefore$ Hence, in this study,
        \begin{itemize}
            \item Premature: $<$ 37 weeks
            \item Normal: $\ge$ 37 weeks
        \end{itemize}
    \end{frame}

    \section{Materials}
    \begin{frame}
        \frametitle{16S rRNA Sequencing}

        \textbf{16S rRNA sequencing} is the \textit{reference method} for bacterial taxonomy \& identification \cite{16S1}

        Reasons \cite{16S2}:
        \begin{itemize}
            \item 16S rRNA exists in almost all bacteria
            \item Functions of the 16S rRNA has not changed over time
            \item 16S rRNA is large enough for bioinformatics
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Train/Test Data vs. Validate Data}
        \begin{itemize}
            \item JBNU/Helixco data
            \begin{itemize}
                \item First data / Second data
                \item Stool data
            \end{itemize}

            \item External data
            \begin{itemize}
                \item The European Bioinformatics Institute (EBI data)
                \item NIH Human Microbiome Project (HMP data)
            \end{itemize}
        \end{itemize}

        \begin{table}
            \centering
            \caption{Sample Information}
            \input{tables/metadata.tex}
        \end{table}
    \end{frame}

    \section{Literature Survey}
    \subsection{EBI Data \protect\cite{validate1}}
    \begin{frame}[allowframebreaks]
        \frametitle{EBI Data}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/Literature/EBI/title.png}
        \end{figure}
        \newpage

        \begin{itemize}
            \item Study Objectives
            \begin{enumerate}
                \item Compare Vaginally vs. Cesarean-section (C-section)
                \item Restore the microbiota of C-section
            \end{enumerate}

            \item Microbial restoration procedure
            \begin{enumerate}
                \item Measure maternal vaginal pH
                \item Put sterile gauze with saline solution in vagina for 1 hour
                \item Swab the infant with the gauze
            \end{enumerate}

            \item Sample collection procedure
            \begin{enumerate}
                \item Sample at right after birth, day 3 and weekly for the first month
                \item Sample from oral, forehead, arm, foot and anal
            \end{enumerate}

            \item Notable Methods/Results
            \begin{enumerate}
                \item Using distance methods: e.g. UniFrac distance, Hamming distance
            \end{enumerate}
        \end{itemize}
    \end{frame}

    \subsection{HMP Data \protect\cite{validate2}}
    \begin{frame}[allowframebreaks]
        \frametitle{HMP Data}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/Literature/HMP/title.png}
        \end{figure}
        \newpage

        \begin{itemize}
            \item Study Objectives
            \begin{enumerate}
                \item Predicting \& Preventing premature
                \item Report community resources
                \item Provide an analysis of the longitudinal, comprehensive, multi-omic profiling of vaginal samples
            \end{enumerate}

            \item Sample collection Procedure
            \begin{enumerate}
                \item Premature birth vs. Matched normal birth
                \item Ethnically diverse cohort
            \end{enumerate}

            \item Notable Methods/Results
            \begin{enumerate}
                \item Imitate figures
            \end{enumerate}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/Literature/HMP/gestational.png}
            \caption{Microbiome Composition during Pregnancy}
        \end{figure}
    \end{frame}

    \section{Methods}
    \subsection{Qiime 2 Workflow}
    \begin{frame}
        \frametitle{Qiime 2 Workflow}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/qiime.png}
            \caption{QIIME 2 workflow \protect\cite{qiime1, qiime2, qiime3}}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Filitering with Quality Score}

        Drawback between:
        \begin{itemize}
            \item Longer sequence read
            \item Higher quality value
        \end{itemize}

        $\therefore$ Select the maximum length $n$ where:
        \begin{equation}
            \begin{array}{c}
                \forall n_i \in \{ n_k | \mbox{MedianQualityScore} \geq 30 \} \\
                \exists ! n \in \{n_i\} : n \geq n_i
            \end{array}
        \end{equation}
    \end{frame}

    \begin{frame}
        \frametitle{Denoising Techniques}

        \begin{itemize}
            \item DADA2: Amplicon Sequence Variants (ASVs) \cite{DADA2}
            \item Deblur: Operational Taxonomic Units (OTUs) \cite{Deblur1}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.4 \linewidth]{figures/tikz/denoising.pdf}
            \caption{Denoising Algorithms}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Taxonomy Classification}

        \begin{itemize}
            \item Greengenes (GG) \cite{greengenes1}
            \item SILVA \cite{silva1, silva2}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.4 \linewidth]{figures/tikz/taxonomy.pdf}
            \caption{Taxonomy Classifications}
        \end{figure}

        “A \textbf{higher} performance at taxonomic levels above \textit{genus level}; \\
        but performance appears to \textbf{drop} at \textit{species level}” \cite{performance1}
    \end{frame}

    \begin{frame}
        \frametitle{Merging Denoising/Taxonomy}

        Merging multiple IDs (ASVs or OTUs) into one, which have
        \begin{itemize}
            \item Different IDs
            \item Identified as same taxonomy
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/tikz/merging.pdf}
            \caption{Example Diagram for Merging Denoising/Taxonomy}
        \end{figure}
    \end{frame}

    \subsection{Abundance Test}
    \begin{frame}
        \frametitle{ANCOM}

        \begin{itemize}
            \item Analysis of composition of microbiome \cite{ANCOM1}
            \item ANCOM detects significantly abundant taxa, while maintain high statistical power
            \item Find taxa that can divide each classes
        \end{itemize}
    \end{frame}

    \subsection{Diversity Indices}
    \begin{frame}
        \frametitle{Diversity Indices}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/phylogenic.jpg}
            \caption{Three dimensions of phylogenic information \protect\cite{phylogenetic1}}
        \end{figure}

        \begin{itemize}
            \item A quantitative measure that shows richness, divergence, and regularity \cite{phylogenetic1}
            \item Alpha diversity indices: the richness of taxa \textbf{at a single community}
            \item Beta diversity indices: the taxonomic differentiation \textbf{between communities}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Alpha  Diversity Indices}

        \begin{itemize}
            \item Evenness index
            \item Faith's Phylogenetic Diversity (Faith PD) index
            \item Oberseved Features index
            \item Shannon's Diversity index
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Beta Diversity Indices}

        \begin{itemize}
            \item Bray-Curtis distance index
            \item Jaccard distnace index
            \item Unweighted UniFrac distance index
            \item Weighted UniFrac distance index
        \end{itemize}
    \end{frame}

    \subsection{Miscellaneous}
    \begin{frame}
        \frametitle{t-distributed Stochastic Neighbor Embedding (t-SNE)}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/tsne.png}
            \caption{t-SNE with handwritten data \protect\cite{tsne1}}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Python Packages}

        \begin{itemize}
            \item Pandas \cite{pandas1}
            \item Scikit-Learn \cite{sklearn1}
            \item SciPy \cite{scipy1}
            \item Matplotlib \cite{matplotlib1}
            \item Seaborn \cite{seaborn1}
        \end{itemize}
    \end{frame}

    \section{Results}
    \subsection{Filtering Results}

    \begin{frame}
        \frametitle{Quality Score from First Data}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.45 \linewidth]{figures/QualityFilter/FirstForward.png}
                &
                \includegraphics[width=0.45 \linewidth]{figures/QualityFilter/FirstReverse.png}
                \\

                \mbox{(a) Forward} & \mbox{(b) Reverse} \\
            \end{array}$
            \caption{Sequence Quality Plot from Helixco Data}
        \end{figure}
        Maximum Length: $n_{Forward} = 300$, $n_{Reverse} = 265$
    \end{frame}

    \begin{frame}
        \frametitle{Quality Score from Second Data}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.45 \linewidth]{figures/QualityFilter/SecondForward.png}
                &
                \includegraphics[width=0.45 \linewidth]{figures/QualityFilter/SecondReverse.png}
                \\

                \mbox{(a) Forward} & \mbox{(b) Reverse} \\
            \end{array}$
            \caption{Sequence Quality Plot from Helixco Data}
        \end{figure}
        Maximum Length: $n_{Forward} = 300$, $n_{Reverse} = 222$
    \end{frame}

    \begin{frame}
        \frametitle{Quality Score from Stool Data}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.45 \linewidth]{figures/QualityFilter/StoolForward.png}
                &
                \includegraphics[width=0.45 \linewidth]{figures/QualityFilter/StoolReverse.png}
                \\

                \mbox{(a) Forward} & \mbox{(b) Reverse} \\
            \end{array}$
            \caption{Sequence Quality Plot from Stool Data}
        \end{figure}
        Maximum Length: $n_{Forward} = 250$, $n_{Reverse} = 251$
    \end{frame}

    \begin{frame}
        \frametitle{Quality Score with EBI Data}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/QualityFilter/EBIForward.png}
            \caption{Sequence Quality Plot from EBI Data}
        \end{figure}
        Maximum Length: $n=150$
    \end{frame}

    \begin{frame}
        \frametitle{Quality Score with HMP Data}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/HMPForward.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/HMPReverse.png}
                \\

                \mbox{(a) Forward} & \mbox{(b) Reverse} \\
            \end{array}$
            \caption{Sequence Quality Plot from HMP Data}
        \end{figure}
        Maximum Length: $n_{forward} = 278$, $n_{Reverse} = 226$
    \end{frame}

    \subsection{Comparing Data}
    \begin{frame}
        \frametitle{Workflow for Comparing Data}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/tikz/brief.pdf}
            \caption{Workflow of t-SNE for Brief Information}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Intersected Taxa}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.2 \linewidth]{figures/tikz/brief/DADA2.gg.pdf}
                &
                \includegraphics[width=0.2 \linewidth]{figures/tikz/brief/DADA2.silva.pdf}
                \\
                \mbox{(a) DADA2 + GG} & \mbox{(b) DADA2 + SILVA} \\

                \includegraphics[width=0.2 \linewidth]{figures/tikz/brief/Deblur.gg.pdf}
                &
                \includegraphics[width=0.2 \linewidth]{figures/tikz/brief/Deblur.silva.pdf}
                \\
                \mbox{(c) Deblur + GG} & \mbox{(d) Deblur + SILVA} \\
            \end{array}$
            \caption{Intersected Taxa Information}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{t-SNE for Comparing Data}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.2 \linewidth]{figures/TSNE/Brief/DB.DADA2.gg.png}
                &
                \includegraphics[width=0.2 \linewidth]{figures/TSNE/Brief/DB.DADA2.silva.png}
                \\
                \mbox{(a) DADA2 + GG} & \mbox{(b) DADA2 + SILVA} \\

                \includegraphics[width=0.2 \linewidth]{figures/TSNE/Brief/DB.Deblur.gg.png}
                &
                \includegraphics[width=0.2 \linewidth]{figures/TSNE/Brief/DB.Deblur.silva.png}
                \\
                \mbox{(c) Deblur + GG} & \mbox{(d) Deblur + SILVA} \\
            \end{array}$
            \caption{t-SNE for Comparing Data}
        \end{figure}
    \end{frame}

    \subsection{t-SNE with Site/Premature Information}
    \begin{frame}
        \frametitle{Workflow for t-SNE with Site/Premature Information}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/tikz/site.pdf}
            \caption{Workflow of t-SNE for Site/Premature Information}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{t-SNE with Site Information}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Site/site.DADA2.gg.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Site/site.DADA2.silva.png}
                \\
                \mbox{(a) DADA2 + GG} & \mbox{(b) DADA2 + SILVA} \\
            \end{array}$
            \caption{t-SNE with Site}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Site/site.Deblur.gg.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Site/site.Deblur.silva.png}
                \\
                \mbox{(c) Deblur + GG} & \mbox{(d) Deblur + SILVA} \\
            \end{array}$
            \caption{t-SNE with Site}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{t-SNE with Premature Information}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Premature/premature.DADA2.gg.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Premature/premature.DADA2.silva.png}
                \\
                \mbox{(a) DADA2 + GG} & \mbox{(b) DADA2 + SILVA} \\
            \end{array}$
            \caption{t-SNE with Site + Premature}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Premature/premature.Deblur.gg.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/TSNE/Premature/premature.Deblur.silva.png}
                \\
                \mbox{(c) Deblur + GG} & \mbox{(d) Deblur + SILVA} \\
            \end{array}$
            \caption{t-SNE with Site + Premature}
        \end{figure}
    \end{frame}

    \subsection{ANCOM}
    \begin{frame}
        \frametitle{Bacterial Abundance Test with ANCOM}
    \end{frame}

    \subsection{Alpha-Diversity}
    \begin{frame}
        \frametitle{Alpha-Diversity}
    \end{frame}

    \subsection{Beta-Diversity}
    \begin{frame}
        \frametitle{Beta-Diversity}
    \end{frame}

    \subsection{Classification}
    \begin{frame}
        \frametitle{Workflow for Classification}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/tikz/classification.pdf}
            \caption{Workflow with Classification}
        \end{figure}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Random Forest Classifier}

        Input Data was treated with \textbf{Deblur} and \textbf{SILVA}.

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/case1/importances.png}
            \caption{Feature Importance derived by Random Forest Classifier}
        \end{figure}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/RandomForest/case1/scores.png}
            \caption{Number of Features vs. Accuracy}
        \end{figure}

        \begin{enumerate}
            \item \textit{Bacteria Firmicutes Bacilli Lactobacillales Lactobacillaceae Lactobacillus Lactobacillus iners}
            \item \textit{Bacteria Fusobacteriota Fusobacteriia Fusobacteriales Leptotrichiaceae Leptotrichia}
            \item \textit{Bacteria Actinobacteriota Actinobacteria}
            \item \textit{Bacteria Firmicutes Bacilli Lactobacillales Lactobacillaceae Lactobacillus}
            \item \textit{Bacteria Firmicutes Clostridia Peptostreptococcales-Tissierellales Peptostreptococcaceae Romboutsia}
            \item \textit{Bacteria Firmicutes Bacilli Mycoplasmatales Mycoplasmataceae Ureaplasma}
            \item \textit{Bacteria Actinobacteriota Actinobacteria Corynebacteriales Corynebacteriaceae Corynebacterium Corynebacterium matruchotii}
        \end{enumerate}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/RandomForest/case1/tree.png}
            \caption{Random Forest Classifier}
        \end{figure}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/RandomForest/case1/feature_0.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/RandomForest/case1/feature_3.png}
                &
                \includegraphics[width=0.3 \linewidth]{figures/RandomForest/case1/feature_4.png}
                \\

                \mbox{(a) \textit{Lactobacillus iners}} & \mbox{(b) \textit{Lactobacillus}} & \mbox{(c) \textit{Romboutsia}}
            \end{array}$
            \caption{Violin Plot of Taxonomy}
        \end{figure}

        \begin{enumerate}[a]
            \item \textit{Bacteria Firmicutes Bacilli Lactobacillales Lactobacillaceae Lactobacillus Lactobacillus iners}
            \item \textit{Bacteria Firmicutes Bacilli Lactobacillales Lactobacillaceae Lactobacillus}
            \item \textit{Bacteria Firmicutes Clostridia Peptostreptococcales-Tissierellales Peptostreptococcaceae Romboutsia}
        \end{enumerate}
    \end{frame}

    \begin{frame}
        \frametitle{\textit{Lactobacillus (Lb.)}}

        \begin{itemize}
            \item Vaginal \textit{Lb.} may be clinically useful tools at PTB under 33 weeks. \cite{lb1}
            \item Presence of \textit{Lb.} sp (odds ratio 0.2) was negatively associated. \cite{lb2}
            \item \textit{Lb. crispatus/gasseri} could decrease the risk of PTB. \cite{lb3}
            \item \textit{Lb.} were associated with decreased risk of PTB. \cite{lb4}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{\textit{Romboutsia}}
    \end{frame}

   	\begin{frame}[allowframebreaks]
        \frametitle{References}
        \bibliographystyle{apacite}
        \bibliography{reference}
    \end{frame}
\end{document}