% !TeX spellcheck = en_US
% !TeX encoding = UTF-8
\documentclass{beamer}

\mode<presentation> { \usetheme{Madrid} }

\usepackage{graphicx, graphics}
\usepackage[notocbib]{apacite}
\usepackage[style=iso]{datetime2}
\usepackage{enumerate}
\DeclareGraphicsExtensions{.pdf, .png, .jpg, .gif}

\AtBeginSection[]
{
    \begin{frame}
        \vfill
        \centering
        \begin{beamercolorbox}[sep=8pt,center,shadow=true,rounded=true]{title}
            \usebeamerfont{title}
            \insertsectionhead
            \par
        \end{beamercolorbox}
        \vfill
    \end{frame}
}

\AtBeginSubsection[]
{
    \begin{frame}
        \vfill
        \centering
        \begin{beamercolorbox}[sep=8pt, center, shadow=true, rounded=true]{title}
            \usebeamerfont{title}
            \insertsectionhead
            \par
        \end{beamercolorbox}
        \begin{beamercolorbox}[sep=4pt, center, shadow=true, rounded=true]{title}
            \usebeamerfont{subtitle}
            \insertsubsectionhead
            \par
        \end{beamercolorbox}
        \vfill
    \end{frame}
}

\title[PTB]{Metagenome Analysis of Premature Birth}

\author[Jaewoong Lee]
{
    Jaewoong Lee
    \and
    Semin Lee
}

\institute[UNIST BME]
{
    Department of Biomedical Engineering
    \newline
    Ulsan National Institute of Science and Technology
    \medskip
    \newline
    \textit{jwlee230@unist.ac.kr}
}

\date{\today}

\begin{document}
    \begin{frame}
        \titlepage
    \end{frame}

	\begin{frame}
        \frametitle{Overview}
        \tableofcontents[hideallsubsections]
    \end{frame}

    \section{Introduction}
    \begin{frame}
        \frametitle{Microbiome}

        \begin{itemize}
            \item Microbiota: the microorganisms which live inside \& on humans \cite{micro1}
            \item Microbiome: $10^{13}$ to $10^{14}$ microorganisms whose which collective genome \cite{micro2}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.3 \linewidth]{figures/microbiome.jpg}
            \caption{Concept of a core human microbiome \protect \cite{micro1}}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{rRNA}

        \begin{itemize}
            \item Ribosomal RNA
            \item Well-known as a key to phylogeny \cite{rrna1}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Premature Birth (Preterm Birth; PTB)}

        PTB:
        \begin{enumerate}
            \item PTB $<$ 37 GW (Gestational week)
            \item Normal $\ge$ 37 GW
        \end{enumerate}

        Detailed PTB:
        \begin{enumerate}
            \item Extremely PTB $<$ 28 GW
            \item 28 GW $\le$ Very PTB $<$ 32 GW
            \item 32 GW $\le$ Late PTB $<$ 37 GW
            \item Normal $\ge$ 37 GW
        \end{enumerate}

        \cite{premature1, premature2}
    \end{frame}

    \section{Materials}
    \begin{frame}
        \frametitle{16S rRNA Sequencing}

        \textbf{16S rRNA sequencing} is the \textit{reference method} for bacterial taxonomy \& identification \cite{16S1}

        Three main reasons \cite{16S2}:
        \begin{itemize}
            \item 16S rRNA exists in almost all bacteria
            \item Functions of the 16S rRNA has not changed over time
            \item 16S rRNA is large enough for bioinformatics
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Train/Test Data vs. Validate Data}
        \begin{itemize}
            \item JBNU/Helixco data
            \begin{itemize}
                \item First data
                \item Second data
                \item Stool data
            \end{itemize}
        \end{itemize}

        \begin{table}
            \centering
            \caption{Sample Information}
            \input{tables/metadata.tex}
        \end{table}
    \end{frame}

    \section{Methods}
    \subsection{Qiime 2 Workflow}
    \begin{frame}
        \frametitle{Qiime 2 Workflow}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/qiime.png}
            \caption{QIIME 2 workflow \protect\cite{qiime1, qiime2, qiime3}}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Filitering with Quality Score}

        Drawback between:
        \begin{itemize}
            \item Longer sequence read
            \item Higher quality value
        \end{itemize}

        $\therefore$ Select the maximum length $n$ where:
        \begin{equation}
            \begin{array}{c}
                \forall n_i \in \{ n_k | \mbox{MedianQualityScore} \geq 30 \} \\
                \exists ! n \in \{n_i\} : n \geq n_i
            \end{array}
        \end{equation}
    \end{frame}

    \begin{frame}
        \frametitle{Denoising Techniques}

        \begin{itemize}
            \item DADA2: Amplicon Sequence Variants (ASVs) \cite{DADA2}
            \item Deblur: Operational Taxonomic Units (OTUs) \cite{Deblur1}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.4 \linewidth]{figures/tikz/denoising.pdf}
            \caption{Denoising Algorithms}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Taxonomy Classification}

        \begin{itemize}
            \item Greengenes (GG) \cite{greengenes1}
            \item SILVA \cite{silva1, silva2}
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.4 \linewidth]{figures/tikz/taxonomy.pdf}
            \caption{Taxonomy Classifications}
        \end{figure}

        “A \textbf{higher} performance at taxonomic levels above \textit{genus level}; \\
        but performance appears to \textbf{drop} at \textit{species level}” \cite{performance1}
    \end{frame}

    \begin{frame}
        \frametitle{Merging Denoising/Taxonomy}

        Merging multiple IDs (ASVs or OTUs) into one, which have
        \begin{itemize}
            \item Different IDs
            \item Identified as same taxonomy
        \end{itemize}

        \begin{figure}
            \includegraphics[width=0.8 \linewidth]{figures/tikz/merging.pdf}
            \caption{Example Diagram for Merging Denoising/Taxonomy}
        \end{figure}
    \end{frame}

    \subsection{Abundance Test}
    \begin{frame}
        \frametitle{ANCOM}

        \begin{itemize}
            \item Analysis of composition of microbiome \cite{ANCOM1}
            \item ANCOM detects significantly abundant taxa, while maintain high statistical power
            \item Find taxa that can divide each classes
        \end{itemize}
    \end{frame}

    \subsection{Diversity Indices}
    \begin{frame}
        \frametitle{Diversity Indices}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/phylogenic.jpg}
            \caption{Three dimensions of phylogenic information \protect\cite{phylogenetic1}}
        \end{figure}

        \begin{itemize}
            \item A quantitative measure that shows richness, divergence, and regularity \cite{phylogenetic1}
            \item Alpha diversity indices: the richness of taxa \textbf{at a single community}
            \item Beta diversity indices: the taxonomic differentiation \textbf{between communities}
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Alpha  Diversity Indices}

        \begin{itemize}
            \item Evenness index
            \item Faith's Phylogenetic Diversity (Faith PD) index
            \item Oberseved Features index
            \item Shannon's Diversity index
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Beta Diversity Indices}

        \begin{itemize}
            \item Bray-Curtis distance index
            \item Jaccard distnace index
            \item Unweighted UniFrac distance index
            \item Weighted UniFrac distance index
        \end{itemize}
    \end{frame}

    \subsection{Miscellaneous}
    \begin{frame}
        \frametitle{t-distributed Stochastic Neighbor Embedding (t-SNE)}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/tsne.png}
            \caption{t-SNE with handwritten data \protect\cite{tsne1}}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Python Packages}

        \begin{itemize}
            \item Pandas \cite{pandas1}
            \item Scikit-Learn \cite{sklearn1}
            \item SciPy \cite{scipy1}
            \item Matplotlib \cite{matplotlib1}
            \item Seaborn \cite{seaborn1}
            \item Statannot
        \end{itemize}
    \end{frame}

    \section{Results}
    \subsection{Filtering Results}
    \begin{frame}
        \frametitle{Quality Score from JBNU/Helixco Data}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/EverythingForward.png}
                &
                \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/EverythingReverse.png}
                \\
                \mbox{(a) Forward} & \mbox{(b) Reverse} \\
            \end{array}$
            \caption{Quality Score Plot}
        \end{figure}

        \begin{itemize}
            \item $\ell_{Forward} = 300$
            \item $\ell_{Reverse} = 245$
        \end{itemize}
    \end{frame}

    \subsection{t-SNE with Clinical Information}
    \begin{frame}
        \frametitle{Workflow for t-SNE with Site/Premature Information}

        \begin{figure}
            \includegraphics[width=0.6 \linewidth]{figures/tikz/site.pdf}
            \caption{Workflow of t-SNE for Site/Premature Information}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{Used Clinical Information}

        \begin{enumerate}
            \item Gestational Diabetes
            \item Maternal Weight
            \item Maternal Weight Gain
            \item Hypertension
            \item PROM
            \item Antibiotic
        \end{enumerate}
    \end{frame}

    \begin{frame}[allowframebreaks]
        \frametitle{Selected t-SNE Plots}

        \begin{figure}
            \includegraphics[width=0.5 \linewidth]{figures/tSNE/everything.DADA2.gg/Gestational_Diabetes.pdf}
            \caption{t-SNE about Gestational Diabetes}
        \end{figure}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/everything.DADA2.gg/Weight.pdf}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/everything.DADA2.gg/Weight_gain.pdf}
                &
                \includegraphics[width=0.3 \linewidth]{figures/tSNE/everything.DADA2.gg/Too_much_weight_gain.pdf}
                \\
                \mbox{(a) Overweight} & \mbox{(b) Weight gain} & \mbox{(c) Too much gain} \\
            \end{array}$
            \caption{t-SNE about Maternal Weight}
        \end{figure}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/tSNE/everything.DADA2.gg/Hypertension.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/tSNE/everything.DADA2.gg/PROM.pdf}
                \\
                \mbox{(a) Hypertension} & \mbox{(b) PROM} \\
            \end{array}$
            \caption{t-SNE about Disease}
        \end{figure}

        \begin{figure}
            $\begin{array}{ccc}
                \includegraphics[width=0.4 \linewidth]{figures/tSNE/everything.DADA2.gg/Mother_Antibiotics.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/tSNE/everything.DADA2.gg/Neonate_Antibiotics.pdf}
                \\
                \mbox{(a) Mother} & \mbox{(b) Neonate} \\
            \end{array}$
            \caption{t-SNE about Antibiotics}
        \end{figure}
    \end{frame}

    \subsection{Bacterial Abundance Test with ANCOM}
    \begin{frame}
        \frametitle{ANCOM?}

        \begin{itemize}
            \item Analysis of composition of microbiomes
            \item ANCOM can be used for analyzing the composition of microbiomes in multiple populations \cite{ANCOM1}
            \item Differential abundance testing
        \end{itemize}

        \begin{enumerate}
            \item clr: Centered $\log (Ratio)$
            \item W: a count of the number of sub-hypothesis which have passed for given species
        \end{enumerate}
    \end{frame}

    \begin{frame}
        \frametitle{ANCOM with ...}

        Site selection:
        \begin{enumerate}
            \item Neonatal mouth: 1-day, 3-day, and 5-day
            \item Cervix
            \item Maternal mouth
            \item Vagina
        \end{enumerate}

        PTB:
        \begin{enumerate}
            \item Premature
            \item Detail premature
        \end{enumerate}
    \end{frame}

    \begin{frame}
        \frametitle{ANCOM with Neonatal Mouth}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Baby.pseudocount.PTB.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Baby.pseudocount.Detail_PTB.pdf}
                \\
                \mbox{(a) PTB} & \mbox{(b) Detailed PTB} \\
            \end{array}$
            \caption{ANCOM with Neonatal Mouth}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{ANCOM with Maternal Mouth}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Mouth.pseudocount.PTB.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Mouth.pseudocount.Detail_PTB.pdf}
                \\
                \mbox{(a) PTB} & \mbox{(b) Detailed PTB} \\
            \end{array}$
            \caption{ANCOM with Maternal Mouth}
        \end{figure}

        \begin{itemize}
            \item \textit{Bacteria Proteobacteria Alphaproteobacteria Rickettsiales mitochondria} family
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{ANCOM with Cervix}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Cervix.pseudocount.PTB.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Cervix.pseudocount.Detail_PTB.pdf}
                \\
                \mbox{(a) PTB} & \mbox{(b) Detailed PTB} \\
            \end{array}$
            \caption{ANCOM with Cervix}
        \end{figure}
    \end{frame}

    \begin{frame}
        \frametitle{ANCOM with Vagina}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Vagina.pseudocount.PTB.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/ANCOM/everything.DADA2.gg.Vagina.pseudocount.Detail_PTB.pdf}
                \\
                \mbox{(a) PTB} & \mbox{(b) Detailed PTB} \\
            \end{array}$
            \caption{ANCOM with Vagina}
        \end{figure}

        \begin{itemize}
            \item \textit{Bacteria  Proteobacteria Epsilonproteobacteria Campylobacterales Campylobacteraceae Campylobacter} genus
        \end{itemize}
    \end{frame}

    \subsection{Diversity Index}
    \begin{frame}
        \frametitle{Rarefaction from First Data}

        \begin{figure}
            $\begin{array}{cc}
                \includegraphics[width=0.4 \linewidth]{figures/Rarefaction/everything.DADA2.pdf}
                &
                \includegraphics[width=0.4 \linewidth]{figures/Rarefaction/everything.Deblur.pdf}
                \\
                \mbox{(a) DADA2} & \mbox{(b) Deblur} \\
            \end{array}$
            \caption{Rarefaction from the Data}
        \end{figure}

        \begin{itemize}
            \item $min(\ell_{DADA2})$: 1046
            \item $min(\ell_{Deblur})$: 4864
        \end{itemize}
    \end{frame}

    \begin{frame}
        \frametitle{Alpha-Diversity}
    \end{frame}

    \begin{frame}
        \frametitle{Beta-Diversity}
    \end{frame}

   	\begin{frame}[allowframebreaks]
        \frametitle{References}
        \bibliographystyle{apacite}
        \bibliography{reference}
    \end{frame}
\end{document}