% !TeX spellcheck = en_US
% !TeX encoding = UTF-8
\documentclass[11pt, a4paper]{article}
\usepackage{graphics, graphicx}
\usepackage{fancyvrb, enumerate}
\usepackage{amsmath, amssymb, amscd, amsfonts}
\usepackage{geometry}
\usepackage{multirow}
\usepackage{url}
\usepackage{listings, listing}
\usepackage{color}
\usepackage{mathptmx}
\usepackage{apacite}
\usepackage[style=iso]{datetime2}

\geometry
{
    top = 20mm,
    bottom = 20mm,
    left = 20mm,
    right = 20mm
}

\title{Microbiome Premature}
\author{Jaewoong Lee}
\date{\today}

\begin{document}
   	\maketitle
    \newpage

    \tableofcontents
    \listoftables
    \listoffigures
    \newpage

    \section{Introduction}
        \subsection{Microbiome}

        \subsection[rRNA]{Ribosomal RNA}

        \subsection{Premature}

    \section{Materials}
        \subsection{16S rRNA Sequencing}

    \section{Methods}
        \subsection{QIIME 2}
            QIIME 2 is a next-generation microbiome bioinformatics platform which is extensible, free, open-source, and community developed \cite{qiime1, qiime2, qiime3}.

            \begin{figure}[htbp]
                \centering
                \includegraphics[width=0.6 \linewidth]{figures/qiime.png}
                \caption{Workflow of QIIME2}
                \label{fig:qiime2}
            \end{figure}

        \subsection{Denoising Techniques}
            \subsubsection{DADA2}
                DADA2 is an open-source software package for modeling and correcting Illumina-sequenced amplicon erros \cite{DADA2}.

            \subsubsection{Deblur}
                Deblur is a software packages which uses error profiles to obtain putative error-free sequences from Illumina MiSeq and HiSeq sequencing platforms \cite{Deblur1}.

        \subsection{Taxonomy Classification}
            \subsubsection{Greengenes}
                Greengenes is a chimera-checked 16S rRNA gene database \cite{greengenes1}.

            \subsubsection{SILVA}
                SILVA is a comprehensive web resource for up-to-date, quality-controlled databases of aligned rRNA gene sequences from the Bacteria domains \cite{silva1, silva2}.

        \subsection{Mothur}
            Mothur is an open-source software package for bioinformatics data processing, especially for the analysis of DNA from microbes \cite{mothur1}.

        \subsection[t-SNE]{t-distributed Stochastic Neighbor Embedding}
            T-distributed stochastic neighbor embedding (t-SNE) visualizeds high-dimensional data by giving each data-point a location in a two-dimensional map \cite{tsne1}.

            \begin{figure}[htbp]
                \centering
                \includegraphics[width=0.5 \linewidth]{figures/tsne.png}
                \caption{t-SNE Visualizations of handwritten digits from MNIST data \protect \cite{tsne1}}
                \label{fig:tsne}
            \end{figure}

        \subsection{Python Packages}
            \subsubsection{Pandas}
                Pandas is a Python library of rich data structure and tools for working with structured data sets \cite{pandas1}.

            \subsubsection{Scikit-Learn}
                Scikit-learn is a Python module which integrating a wide range of state-of-the-art machine learning algorithms for medium-scale supervised and unsupervised problems \cite{sklearn1}.

            \subsubsection{Matplotlib}
                Matplotlib is a two-dimensional graphics package used for Python for image generation \cite{matplotlib1}.

            \subsubsection{Seaborn}
                Seaborn is a Python data visualization library based on Matplotlib \cite{seaborn1}.


    \section{Results}

    \section{Discussion}

    \bibliographystyle{apacite}
    \bibliography{reference}
\end{document}