% !TeX spellcheck = en_US
% !TeX encoding = UTF-8
\documentclass[a4paper]{article}
\usepackage{graphics, graphicx}
\usepackage{fancyvrb, enumerate}
\usepackage{amsmath, amssymb, amscd, amsfonts}
\usepackage{geometry}
\usepackage{multirow}
\usepackage{url}
\usepackage{listings, listing}
\usepackage{color}
\usepackage{mathptmx}
\usepackage[numberedbib]{apacite}
\usepackage[style=iso]{datetime2}

\geometry
{
    top = 20mm,
    bottom = 20mm,
    left = 20mm,
    right = 20mm
}

\title{Microbiome Premature}
\author{
    Jaewoong Lee
    \and
    Semin Lee
}
\date{\today}

\begin{document}
   	\maketitle
    \newpage

    \tableofcontents
    \listoftables
    \listoffigures
    \newpage

    \section{Introduction}
        \subsection{Microbiome}
            After the Human Genome Project was finished, the microorganisms which live along humans, as known as the microbiota, are considered overwhelmed human cells \cite{micro1}. Moreover, the microbiome, the collective genome from these microbiota \cite{micro2}, serve as the trait of individual have not to evolve on their own \cite{micro1}. Furthermore, human microbiome is effected by host's life style as figure \ref{fig:microbiome}.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.4 \linewidth]{figures/microbiome.jpg}
                \caption{Concept of a core human microbiome. \protect \cite{micro1}}
                \label{fig:microbiome}
            \end{figure}

        \subsection[rRNA]{Ribosomal RNA}
            Ribosomal RNA (rRNA) plays the main roles in a cell. This main roles include mRNA selection, tRNA binding, proof-reading, factor binding, and \textit{et cetera} \cite{rrna2}. Because of its momentous roles, rRNA could be preserved amongst whole bacteria throughout the evolution.

        \subsection{Premature}
            Premature (PTB; stands for Preterm Birth) is the birth of a baby earlier than 37 gestational weeks, as Figure \ref{fig:ptb} \cite{premature1}. Premature infants have more risk such as hearing problems and sight problems.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.4 \linewidth]{figures/premature.png}
                \caption{Definition of Premature \protect \cite{premature1}}
                \label{fig:ptb}
            \end{figure}

    \section{Materials}
        \subsection{16S rRNA Sequencing}
            rRNA has been kept among bacteria; thus, 16S rRNA exists in almost bacteria, and its functions has not changed over time. Also, 16S rRNA is large enough for bioinformatics \cite{16S2}. Hence, 16S rRNA sequencing is the reference method for bacterial taxonomy classification and identification \cite{16S1}.

            There are three databases which for machine learning: Helixco data, EBI data, and HMP data. Metadata of these databases is as table \ref{tb:metadata}.

            \begin{table}[p]
                \centering
                \caption{Metadata of Data}
                \label{tb:metadata}
                \input{tables/metadata.tex}
            \end{table}

            \subsubsection{JBNU/Helixco Data}

            \subsubsection[EBI Data]{European Bioinformatics Institute Data}
                EBI data was collected by European Bioinformatics Institute (EBI) \cite{validate1}. EBI data aimed to compare Cesarean section birth and vaginal birth; thus, every participants in EBI data is on term, not PTB.

            \subsubsection[HMP Data]{Human Microbiome Project Data}
                HMP data was collected by Human Microbiome Project (HMP) \cite{validate2}. HMP data aimed to compare PTB and on-term birth; thus, every participants in HTMP data is PTB.

    \section{Methods}
        \subsection{Docker}
            Docker is light-weight Linux containers for consistent development and deployment \cite{docker1}.

        \subsection{QIIME 2}
            QIIME 2 is a next-generation microbiome bioinformatics platform which is extensible, free, open-source, and community developed \cite{qiime1, qiime2, qiime3}.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.6 \linewidth]{figures/qiime.png}
                \caption{Workflow of QIIME2}
                \label{fig:qiime2}
            \end{figure}

        \subsection{Denoising Algorithms}
            There are two denoising algorithms which are provided by QIIME as figure \ref{fig:denoising}: DADA2 \cite{DADA2} and Deblur \cite{Deblur1}.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.3 \linewidth]{figures/tikz/denoising.pdf}
                \caption{Denoising Algorithms}
                \label{fig:denoising}
            \end{figure}

            \subsubsection{DADA2}
                DADA2 is an open-source software package for modeling and correcting Illumina-sequenced amplicon errors \cite{DADA2}.

            \subsubsection{Deblur}
                Deblur is a software packages which uses error profiles to obtain putative error-free sequences from Illumina MiSeq and HiSeq sequencing platforms \cite{Deblur1}.

        \subsection{Taxonomy Classification Algorithms}
            There are two taxonomy classification algorithms which are provided by QIIME as figures \ref{fig:taxonomy}: Greengenes \cite{greengenes1} ans SILVA \cite{silva1, silva2}.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.3 \linewidth]{figures/tikz/taxonomy.pdf}
                \caption{Taxonomy Classification Algorithms}
                \label{fig:taxonomy}
            \end{figure}

            \subsubsection{Greengenes}
                Greengenes (GG) is a chimera-checked 16S rRNA gene database \cite{greengenes1}.

            \subsubsection{SILVA}
                SILVA is a comprehensive web resource for up-to-date, quality-controlled databases of aligned rRNA gene sequences from the Bacteria domains \cite{silva1, silva2}.

        \subsection{Merging Taxonomy}
            After applying denoising algorithms and taxonomy classification algorithms, some reads have different IDs (ASV OTU), but are identified as same taxonomy. In that cases, these reads are merged into same taxonomy as figure \ref{fig:merging}.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.5 \linewidth]{figures/tikz/merging.pdf}
                \caption{Example Diagram for Merging Taxonomy}
                \label{fig:merging}
            \end{figure}

        \subsection[t-SNE]{t-distributed Stochastic Neighbor Embedding}
            T-distributed stochastic neighbor embedding (t-SNE) visualizes high-dimensional data by giving each data-point a location in a two-dimensional map \cite{tsne1}.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.5 \linewidth]{figures/tsne.png}
                \caption{t-SNE Visualizations of handwritten digits from MNIST data \protect \cite{tsne1}}
                \label{fig:tsne}
            \end{figure}

        \subsection{Python Packages}
            \subsubsection{Pandas}
                Pandas is a Python library of rich data structure and tools for working with structured data sets \cite{pandas1}.

            \subsubsection{Scikit-Learn}
                Scikit-learn is a Python module which integrating a wide range of state-of-the-art machine learning algorithms for medium-scale supervised and unsupervised problems \cite{sklearn1}.

            \subsubsection{Matplotlib}
                Matplotlib is a two-dimensional graphics package used for Python for image generation \cite{matplotlib1}.

            \subsubsection{Seaborn}
                Seaborn is a Python data visualization library based on Matplotlib \cite{seaborn1}.

    \section{Results}
        \subsection{Deciding Trimming Length}
            Deblur require filtering step; though, DADA2 contains filtering step. For filtering step, trimming length should be decided. In other words, the sequence which is longer than specific length should be removed, because of the quality of sequences. There is no canonically admitted methods for deciding trimming length; thus, trimming length $n$ would be decided as equation \ref{eq:trim}. Hence, trimming lengths are decided as shown as table \ref{tb:trim}.

            \begin{equation}
                \begin{array}{c}
                    \forall n_i \in \{ n_k | \mbox{MedianQualityScore} \geq 30 \} \\
                    \exists ! n \in \{n_i\} : n \geq n_i
                \end{array}
                \label{eq:trim}
            \end{equation}

            \begin{figure}[p]
                \centering
                $\begin{array}{cc}
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/FirstForward.png}
                    &
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/FirstReverse.png}
                    \\
                    \mbox{(a) Forward} & \mbox{(b) Reverse}
                \end{array}$
                \caption{Sequence Quality Plot from First JBNU/Helixco Data}
                \label{fig:quality-first}
            \end{figure}

            \begin{figure}[p]
                \centering
                $\begin{array}{cc}
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/SecondForward.png}
                    &
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/SecondReverse.png}
                    \\
                    \mbox{(a) Forward} & \mbox{(b) Reverse}
                \end{array}$
                \caption{Sequence Quality Plot from Second JBNU/Helixco Data}
                \label{fig:quality-second}
            \end{figure}

            \begin{figure}[p]
                \centering
                $\begin{array}{cc}
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/StoolForward.png}
                    &
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/StoolReverse.png}
                    \\
                    \mbox{(a) Forward} & \mbox{(b) Reverse}
                \end{array}$
                \caption{Sequence Quality Plot from Stool JBNU/Helixco Data}
                \label{fig:quality-stool}
            \end{figure}

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/EBIForward.png}
                \caption{Sequence Quality Plot from EBI Data}
                \label{fig:quality-EBI}
            \end{figure}

            \begin{figure}[p]
                \centering
                $\begin{array}{cc}
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/HMPForward.png}
                    &
                    \includegraphics[width=0.4 \linewidth]{figures/QualityFilter/HMPReverse.png}
                    \\
                    \mbox{(a) Forward} & \mbox{(b) Reverse}
                \end{array}$
                \caption{Sequence Quality Plot from HMP Data}
                \label{fig:quality-HMP}
            \end{figure}

            \begin{table}[p]
                \centering
                \caption{Trimming Lengths of Databases}
                \label{tb:trim}
                \begin{tabular}{c | c c}
                    & Sequence Quality Plot & Trimming Length \\ \hline
                    Helixco Data & 265 & Figure \ref{fig:quality-first} \\
                    EBI Data & 150 & Figure \ref{fig:quality-EBI} \\
                    HMP Data & 226 & Figure \ref{fig:quality-HMP}
                \end{tabular}
            \end{table}

        \subsection{t-SNE for Comparing Databases}
            To compare three databases, workflow, which as figure \ref{fig:workflow-brief}, was executed:
            \begin{enumerate}
                \item Select intersected taxonomies that means the taxonomy which can be founded in every database.
                \item Draw t-SNE plot in 2-dimension with intersected taxonomies.
            \end{enumerate}

            Thereupon, intersected taxonomies are as figure \ref{fig:inter-brief}. There are around 200 - 400 intersected taxa. With these intersected taxa, t-SNE plot was derived as figure \ref{fig:tsne-brief}.

            \begin{figure}[p]
                \centering
                \includegraphics[width=0.3 \linewidth]{figures/tikz/brief.pdf}
                \caption{Workflow of t-SNE for Brief Information}
                \label{fig:workflow-brief}
            \end{figure}

            \begin{figure}[p]
                \centering
                $\begin{array}{cc}
                    \includegraphics[width=0.3 \linewidth]{figures/tikz/brief/DADA2.gg.pdf}
                    &
                    \includegraphics[width=0.3 \linewidth]{figures/tikz/brief/DADA2.silva.pdf}
                    \\
                    \mbox{(a) DADA2 + GG} & \mbox{(b) DADA2 + SILVA} \\

                    \includegraphics[width=0.3 \linewidth]{figures/tikz/brief/Deblur.gg.pdf}
                    &
                    \includegraphics[width=0.3 \linewidth]{figures/tikz/brief/Deblur.silva.pdf}
                    \\
                    \mbox{(c) Deblur + GG} & \mbox{(d) Deblur + SILVA}
                \end{array}$
                \caption{Count of Intersected Taxonomies}
                \label{fig:inter-brief}
            \end{figure}

            \begin{figure}[p]
                \centering
                $\begin{array}{cc}
                    \includegraphics[width=0.3 \linewidth]{figures/TSNE/Brief/DB.DADA2.gg.png}
                    &
                    \includegraphics[width=0.3 \linewidth]{figures/TSNE/Brief/DB.DADA2.silva.png}
                    \\
                    \mbox{(a) DADA2 + GG} & \mbox{(b) DADA2 + SILVA} \\

                    \includegraphics[width=0.3 \linewidth]{figures/TSNE/Brief/DB.Deblur.gg.png}
                    &
                    \includegraphics[width=0.3 \linewidth]{figures/TSNE/Brief/DB.Deblur.silva.png}
                    \\
                    \mbox{(c) Deblur + GG} & \mbox{(d) Deblur + SILVA}
                \end{array}$
                \caption{t-SNE for Comparing Databases}
                \label{fig:tsne-brief}
            \end{figure}

    \section{Discussion}
        \subsection{t-SNE for Comparing Databases}

    \bibliographystyle{apacite}
    \bibliography{reference}
\end{document}